crates/subspace-core-primitives/src/crypto.rs

```diff
@@ -211,6 +211,24 @@ impl TryFrom<[u8; Self::FULL_BYTES]> for Scalar {
     }
 }
 
+impl TryFrom<&Blake3Hash> for Scalar {
+    type Error = String;
+
+    #[inline]
+    fn try_from(value: &Blake3Hash) -> Result<Self, Self::Error> {
+        Self::try_from(value.0)
+    }
+}
+
+impl TryFrom<Blake3Hash> for Scalar {
+    type Error = String;
+
+    #[inline]
+    fn try_from(value: Blake3Hash) -> Result<Self, Self::Error> {
+        Self::try_from(value.0)
+    }
+}
```

Comment:
I see no reason why we need these conversions, `Scalar` and `Blake3Hash` are completely different unrelated things, there should be no widespread use case for this to be needed. You can always get raw bytes out of hash and try to convert it to scalar that way.